src/main/scala/subsystem/HasTiles.scala

```diff
@@ -86,6 +91,49 @@ trait HasTileInterruptSources
   }
 }
 
+/** These are sources of "constants" that are driven into the tile.
+  * While they are not expected to change dyanmically while the tile is executing code,
+  * they may be either tied to a contant value or programmed during boot or reset.
+  * They need to be instantiated before tiles are attached to the subsystem containing them.
+  */
+trait HasTileInputConstants extends InstantiatesTiles { this: BaseSubsystem =>
+  // Collect dynamic hart id prefixes and/or distribute static hart ids to the tiles
+  val tileHartIdNode = BundleBridgeEphemeralNode[UInt]()
+  val tileHartIdNexusNode = BundleBridgeNexus[UInt](
+    inputFn = BundleBridgeNexus.orReduction[UInt](p(HartPrefixKey)) _,
+    outputFn = (prefix: UInt, n: Int) =>  Seq.tabulate(n) { i =>
+      val y = dontTouch(prefix | hartIdList(i).U(p(MaxHartIdBits).W))
+      if (p(HartPrefixKey)) BundleBridgeNexus.safeRegNext(y) else y
+    },
+    default = Some(() => 0.U(p(MaxHartIdBits).W))
+  )
+  // TODO: this also needs to be wired into the DebugModule's hart selection circuit?
+
+  // Collect a reset vector address and distribute to the tiles
+  val tileResetVectorNode = BundleBridgeEphemeralNode[UInt]()
```

Comment:
does this mean that all tiles have the same reset vector? has this always been the case?

Reply:
The only way to get them to have separate reset vectors is to flip the `SubsystemExternalResetVectorKey` flag, which has historically been the only way to accomplish that outcome. In the closed source, flipping the flag would be overridden by Chisel last connect semantics by mixing in different traits adding roms or etc that override the inputs to all be the same. Now you have to choose whether to connect the tiles to the NexusNode (all same) or to the IONodes (all different, unless the parent drives them to be the same).

Reply:
by "flip" you mean "set SubsystemExternalResetVectorKey -> true"? If you set that, then all tiles get separate reset vectors?

Reply:
Yes